functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +88,58 @@ export const processNewAllotment = functions.database
     });
     return 1;
   });
+
+const serialRegex = "^[a-zA-Z]+-\\d+";
```

Comment:
I'm sure we already have a regex for the file names to extract list, serial number and the other parts. Please look around and use that regex. If it doesn't exist then create it, but outside SE module, as it is common to all the modules.